### DISCUSSION TITLE
WebPack 4 + Pug.js é viável?

### DISCUSSION THREAD

**Discussion by allefbr** [frontendbr/forum#1126] [Perguntas e Respostas]
Olá meus queridos, 

Estou com uma dúvida em relação ao WebPack 4 + Pug.js, gostaria de saber como vocês trabalham com isso e se não qual a melhor opção para templates engine's. 

Grato!

**Comment by rodrigooler** [comment]
Depende da sua necessidade, mas webpack 4 + pug.js parece ser viável sim, é um excelente template engine.

**Comment by WillianSugiyama** [comment]
Estou com um projeto no qual estou usando PUG.js + webpack4, meu único problema com essa stack é a parada de múltiplos HTML's. Mas é uma stack rápida e fácil !

**Comment by allefbr** [comment]
Comecei a fazer, mas estou enfrentando uns problemas para importar JSON para o projeto e com @imports do SASS, por isso da pergunta, saber como estão trabalhando e se possível compartilhar os starts de vocês. 

Tipo qual loader estão usando para o PUG, e como injetam dados nos templates.

**Comment by guilhermejcgois** [comment]
Aqui usamos o [pug-loader](https://github.com/pugjs/pug-loader): 
```
{
    include: /\.pug/,
    loader: 'pug-loader',
    query: {
        data: data
    },
},
```
também usamos o [HTMLWebpackPlugin](https://github.com/jantimon/html-webpack-plugin):
```
new HtmlWebpackPlugin({
    template: './src/index.pug'
}),
```
no template, os dados seriam acessados a partir de `data`, por exemplo para pegar o `env` seria simplesmente `data.env`

**Comment by allefbr** [comment]
@guilhermejcgois esse `data` que você passa no objeto ele vem de onde? 

Estou recebendo `undefined`, imagino que onde tem o data seria para passar o meu JSON, porém como você usou comoo exemplo o `data.env` fique na dúvida agora.

Outra coisa, você trabalha com Sass também no Webpack? Se sim, poderia me da um exemplo de configuração, estou tento problemas com o `@import`.

**Comment by guilhermejcgois** [comment]
o `data` poderia ser qualquer objeto, no nosso caso fazemos para diferenciar os ambientes de prod/qa/dev e é um valor estático, no projeto que estou trabalhando não temos nada dinâmico... no seu caso o JSON poderia ser qualquer membro do objeto que você escolher, não sei se consegui ser claro rs

Sim, trabalhamos com sass... é `@Important` mesmo ou `@import`? Imagino que seja o segundo e usamo o [sass-loader](https://github.com/webpack-contrib/sass-loader) e não estou importando nada da node_modules se for esse seu problema... Mas me parece ser outro tipo de problema, talvez se você abrir outra issue pro sass seja mais fácil da galera te ajudar

**Comment by allefbr** [comment]
Obrigado pelas dicas irmão, vou da umas olhadas aqui no `sass-loader` e se nada rolar abrirei outra issue.

**Comment by felipefialho** [comment]
Faz e faz bastante.

Alias, a nova versão do [Kratos Boilerplate](https://github.com/LFeh/kratos-boilerplate) que vou tentar soltar nos proximos dias, vai ter exatamente essa stack 😄
